PWGCF/EbyEFluctuations/Tasks/kaonIsospinFluctuations.cxx

```diff
@@ -0,0 +1,1409 @@
+      IdMethod = 0;
+      return true;
+    }
+    if (track.hasTOF() && track.beta() < 0.0 && selProtonTPCInnerParam(track)) {
+      IdMethod = 0;
+      return true;
+    }
+    if (track.hasTOF() && track.beta() > 0.0 && selProtonTPCInnerParam(track)) {
+      IdMethod = 0;
+      return true;
+    }
+    if (track.hasTOF() && track.beta() > 0.0 && !selProtonTPCInnerParam(track)) {
+      IdMethod = 1;
+      return selProtonTOF(track);
+    }
+    return false;
+  }
+
+  // Deuteron
+  template <typename T>
+  bool selDeuteron(T track, int& IdMethod)
+  {
+    if (!track.hasTOF() && selDeuteronTPCInnerParam(track)) {
+      IdMethod = 0;
+      return true;
+    }
+    if (track.hasTOF() && track.beta() < 0.0 && selDeuteronTPCInnerParam(track)) {
+      IdMethod = 0;
+      return true;
+    }
+    if (track.hasTOF() && track.beta() > 0.0 && selDeuteronTPCInnerParam(track)) {
+      IdMethod = 0;
+      return true;
+    }
+    if (track.hasTOF() && track.beta() > 0.0 && !selDeuteronTPCInnerParam(track)) {
+      IdMethod = 1;
+      return selDeuteronTOF(track);
+    }
+    return false;
+  }
+
+  // Electron
+  template <typename T>
+  bool selElectron(T track, int& IdMethod)
+  {
+    if (!track.hasTOF() && selElectronTPCInnerParam(track)) {
+      IdMethod = 0;
+      return true;
+    }
+    if (track.hasTOF() && track.beta() < 0.0 && selElectronTPCInnerParam(track)) {
+      IdMethod = 0;
+      return true;
+    }
+    if (track.hasTOF() && track.beta() > 0.0 && selElectronTPCInnerParam(track)) {
+      IdMethod = 0;
+      return true;
+    }
+    if (track.hasTOF() && track.beta() > 0.0 && !selElectronTPCInnerParam(track)) {
+      IdMethod = 1;
+      return selElectronTOF(track);
+    }
+    return false;
+  }
+
+  template <typename T>
+  int findV0Tag(const T& posDaughterTrack, const T& negDaughterTrack, int& posPiIdMethod, int& posPrIdMethod, int& negPiIdMethod, int& negPrIdMethod)
+  {
+    bool posIsPion = false;
+    bool posIsProton = false;
+    bool negIsPion = false;
+    bool negIsProton = false;
+
+    int v0TagValue = 0;
+
+    // Check if positive track is pion or proton
+    if (selPion(posDaughterTrack, posPiIdMethod))
+      posIsPion = true; // Coming From K0s    -> PiPlus + PiMinus and AntiLambda -> PiPlus + AntiProton
+    if (selProton(posDaughterTrack, posPrIdMethod))
+      posIsProton = true; // Coming From Lambda -> proton + PiMinus
+    if (selPion(negDaughterTrack, negPiIdMethod))
+      negIsPion = true; // Coming From K0s       -> PiPlus + PiMinus and Lambda -> proton + PiMinus
+    if (selProton(negDaughterTrack, negPrIdMethod))
+      negIsProton = true; // Coming From AntiLambda -> PiPlus + AntiProton
+
+    if (posIsPion && negIsPion) {
+      v0TagValue += 1; // It is K0s
+    }
+    if (posIsProton && negIsPion) {
+      v0TagValue += 2; // It is Lambda
+    }
+    if (posIsPion && negIsProton) {
+      v0TagValue += 4; // It is AntiLambda
+    }
```

Comment:
I would suggest using named constants and flags instead
For instance you can use an integer variable as having 32 flags, one per bit, and you use named constants for the meaning of each bit, for instance if you want the second bit to mean that the pair is a Lambda you might use
```
#define kPAIRBIT_ISLAMBDA (1 << 1)
```
you can set the corresponding bits in your variable and check for them on any place in your code
With the current approach you have to check the whole arithmetic